File: 78-Subsets/78-Subsets.java
// Last updated: 7/12/2026, 6:24:01 PM
class Solution {
    public List<List<Integer>> subsets(int[] nums) {
        List<Integer> list = new ArrayList<Integer>();
        List<List<Integer>> res = new ArrayList<>();

        func(nums,0,list,res);

        return res;
    }

    public void func(int[] nums, int index,List<Integer> list, List<List<Integer>> res){

        if(index == nums.length){
            res.add(new ArrayList(list));
            return;
        }

        // pick
        list.add(nums[index]);
        func(nums,index + 1,list,res);

        // not pick
        list.remove(list.size() - 1);
        func(nums,index + 1,list,res);

    }
}

File: 78-Subsets/78-Subsets.py
# Last updated: 7/12/2026, 6:23:52 PM
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:

        n = len(nums)
        subsets = 1 << n
        ans = []

        for i in range(subsets):
            list = []
            for j in range(n):
                if i & (1 << j):
                    list.append(nums[j])

            ans.append(list)

        return ans


            
        